pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -68,15 +70,21 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 
 	// TODO(user): Modify this to be the types you create that are owned by the primary resource
 	// Watch for changes to secondary resource Pods and requeue the owner Infinispan
-	err = c.Watch(&source.Kind{Type: &appsv1beta1.StatefulSet{}}, &handler.EnqueueRequestForOwner{
-		IsController: true,
-		OwnerType:    &infinispanv1.Infinispan{},
-	})
+	err = watchResourceKind(&appsv1beta1.StatefulSet{}, c, err)
```

Comment:
These calls generate `reconcile.Request` for StatefulSet and Service objects, but how they are handled in the `Reconcile` function?

Reply:
Right, I'll remove this parts from the PR. Should be handled for all services with the parallel PR you're working on.